src/Amido.Stacks.DynamoDB/Converters/DynamoDbGuidConverter.cs

```diff
@@ -7,23 +7,29 @@ public class DynamoDbGuidConverter : IPropertyConverter
 	{
 		public DynamoDBEntry ToEntry(object value)
 		{
-			Guid id = Guid.Parse(value?.ToString());
-			DynamoDBEntry entry = new Primitive
+			if (Guid.TryParse(value.ToString(), out Guid id))
 			{
-				Value = id.ToString()
-			};
+				DynamoDBEntry entry = new Primitive
+				{
+					Value = id.ToString()
+				};
 
-			return entry;
+				return entry;
+			}
+			else
+			{
+				throw new ArgumentNullException();
+			}
 		}
 
 		public object FromEntry(DynamoDBEntry entry)
 		{
-			Primitive primitive = entry as Primitive;
+			Primitive primitive = entry.AsPrimitive();
 			if (primitive == null || !(primitive.Value is String) || string.IsNullOrEmpty((string)primitive.Value))
-				throw new ArgumentOutOfRangeException();
+				throw new ArgumentNullException();
 
-			Guid id = Guid.Parse(primitive.Value.ToString());
-			return id;
+			if (Guid.TryParse(primitive.Value.ToString(), out Guid id)) return id;
+			else throw new ArgumentNullException();
```

Comment:
Do we need an `else` here? 

When the `if` condition is triggered it will short circuit this, right?

Reply:
Yes we do. The method won't compile otherwise since we need to return something to satisfy the method signature. In our case it's return a Guid or fail! `Guid.Parse()` throws an exception if it's not a valid Guid, however you have to know that - the behaviour is hidden. That's why I prefer the more explicit `Guid.TryParse()` where you can see everything.